vernac/declare.ml

```diff
@@ -2570,7 +2575,17 @@ let solve_obligation ?check_final prg num tac =
     Proof_ending.End_obligation {name; num; auto; check_final}
   in
   let cinfo = CInfo.make ~name:obl.obl_name ~typ:(EConstr.of_constr obl.obl_type) () in
+
+  (* EJGA/HH: Note that this using won't have any effect as
+     [obligation_terminator] discards the using data in the entry set
+     by the call below to [Proof.start_core]. We still pass it to the
+     proof context, but emit a warning. Note that we only warn if
+     using the attribute, [Proof Using] doesn't do the check to see if
+     we are in a obligation [could be done by checking terminator type
+     tho] *)
   let using =
+    (* TODO: locate this *)
+    let () = warn_unused_using () ?loc:None in
     let using = Internal.get_using prg in
     let env = Global.env () in
     let f {CInfo.name; typ; _} = name, [typ] in
```

Comment:
The warning should be when `using` is non-`None`, and, then, the idea would be that the useless computation that follows is removed.